src/rust/engine/ui/src/display.rs

```diff
@@ -244,7 +244,7 @@ impl EngineDisplay {
 
   fn render_for_pipe(&self) {
     // TODO: Handle non-tty output w polling interval adjustment + summary rendering.
-    panic!("TODO");
+    // Nothing needs to be printed to pipe by default.
```

Comment:
So, while this definitely shouldn't panic, I think that rather than disabling this at this level, we should ensure that when we're inspecting the "should we render the UX" global option, we check whether we're in a tty, and then we explicitly disable the UI rendering if we do not have a TTY.

This information is passed from the pantsd client to the pantsd server via: https://github.com/pantsbuild/pants/blob/master/src/python/pants/java/nailgun_protocol.py#L285-L314 ... but I think that the gist is still that looking at `isatty()` for `stdin/stderr/stdout` should tell you whether to render.

Having done that, we should maybe just remove this method entirely if possible...?

Reply:
Should've clarify a bit that this is not required for this change. Just that I don't want pants to crash on pipe. There is already logic above it which does not call this function at all if it is tty.

https://github.com/pantsbuild/pants/blob/298d821b8565e4bd2e6fe4d78960204de8254861/src/rust/engine/ui/src/display.rs#L261-L268